lib/dsc-lib/src/util.rs

```diff
@@ -232,6 +233,29 @@ pub fn resource_id(type_name: &str, name: &str) -> String {
     result
 }
 
+pub fn canonicalize_which(executable: &str, cwd: Option<&str>) -> Result<String, DscError> {
+    // Use PathBuf to handle path separators robustly
+    let mut executable_path = PathBuf::from(executable);
+    if cfg!(target_os = "windows") && executable_path.extension().is_none() {
+        executable_path.set_extension("exe");
```

Comment:
We should document that we're only going to check for '.exe' if no extension was supplied. This is different than what I believe to be Windows' standard behavior that checks for all of '.cmd', '.bat', variations etc.

Reply:
This was detailed in the PR description